milhoja/cli.py

```diff
@@ -1,30 +1,53 @@
-# -*- coding: utf-8 -*-
+import os
+import sys
+sys.path.append(os.path.join(os.path.dirname(__file__), '..'))  # noqa: E402
+
 
 import json
+import logging
 import click
 from cookiecutter.cli import validate_extra_context
 from cookiecutter.exceptions import CookiecutterException
-from .milhoja import Milhoja # indeed ...
-from .utils import open_repository, open_or_init_repository
-from .errors import MilhojaException
+from milhoja.core import Milhoja
+from milhoja.utils import open_repository, open_or_init_repository
+from milhoja.errors import MilhojaException
+
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+
 
 @click.group()
 @click.option(
-    u'-C',
-    default=u'.',
-    help=u'Run as if milhoja was started in <path> instead of the current working directory.',
+    '-C',
+    default='.',
+    help='Run as if milhoja was started in <path> instead of the current working directory.',
     type=click.Path()
 )
+@click.option(
+    '--verbose',
+    default=False,
+    is_flag=True,
+    help='Enables the debug logging.'
+)
 @click.pass_context
-def main(ctx, c):
+def main(ctx, c, verbose):
     """Script entry point for Milhoja commands.
 
     Arguments:
     ctx -- CLI context.
     c -- Path where to run milhoja
+    verbose -- Enables debug logging
     """
     ctx.obj = dict()
-    ctx.obj['target'] = c
+    ctx.obj.update({
+        'target': c,
+        'verbose': verbose
+    })
+
+    level = logging.DEBUG if verbose else logging.INFO
+    logger.setLevel(level)
```

Comment:
You'll likely want to set the level on your handler `StreamHandler` too, since it will default to `logging.ERROR`.